src/items.cpp

```diff
@@ -326,9 +326,33 @@ bool Items::loadFromXml()
 			parseItemNode(itemNode, id++);
 		}
 	}
+
+	buildInventoryList();
 	return true;
 }
 
+void Items::buildInventoryList()
+{
+	inventory.reserve(30000);
+	for(const auto &type: items) {
+		if(type.weaponType != WEAPON_NONE || type.ammoType != AMMO_NONE ||
+			type.attack != 0 || type.defense != 0 ||
+			type.extraDefense != 0 || type.armor != 0 ||
+			(type.slotPosition & SLOTP_NECKLACE) == SLOTP_NECKLACE ||
```

Comment:
You don't need to compare it to `SLOTP_NECKLACE`, it will coerce to `true` if not equal to zero, which, bitwisely, means it's equal to `SLOTP_NECKLACE` anyways. Same for below.